### DISCUSSION TITLE
Solidworks type navigation addon in the navigation style

### DISCUSSION THREAD

**Discussion by hemangjoshi37a** [FreeCAD/FreeCAD#18635] [Code contribution discussions]
I want to have Solidworks type navigation addon in the navigation style. 
Please if anyone can contribute in this direction .

**Comment by luzpaz** [comment]
Would you be willing to write up a detailed proposal or what needs to be done? 
What would each mouse movement do?
What would each mouse movement in combination with keyboard shortcut do?
What keyboard combo shortcuts are you proposing? 

Organize all this into a document so if/when a dev sees it, they will have all the info they clearly need to map the functionality.

Documentation references:
* https://wiki.freecad.org/Mouse_navigation

All Navigation code is currently in these files in the source code:
https://github.com/FreeCAD/FreeCAD/blob/a7c550da98035a9a33e62a5fd8947f8455fe49a7/src/Gui/CMakeLists.txt#L888-L898

**Reply by hemangjoshi37a** [reply]
Thanks for the quick response. I will prepare it and add here soon. 👍

**Comment by luzpaz** [comment]
In the meantime, I'm going to close the discussion. If you make progress feel free to re-open discussion
